TeamCode/src/main/java/org/firstinspires/ftc/teamcode/PowerPlay/Trajectories.kt

```diff
@@ -0,0 +1,115 @@
+import org.firstinspires.ftc.teamcode.trajectorysequence.TrajectorySequence
+
+data class TrajectorySet(
+    val scoringPos: Pose2d,
+    val navigateTrajectory: TrajectorySequence,
+    val parkTrajectory: TrajectorySequence,
+)
+
+data class TrajectoryParameters(
+    val side: InitialSide,
+    val foundTag: Int,
+)
+
+class Trajectories(
+    drive: SparkyMecanumDrive
+) {
+
+    private val navigationTrajectoryMap: MutableMap<InitialSide, Pair<TrajectorySequence, Pose2d>> =
+        mutableMapOf()
+    private val trajectoryMap: MutableMap<TrajectoryParameters, TrajectorySet> = mutableMapOf()
+
+    init {
+        val possibleTags = listOf(
+            PowerPlayAutoRoadRunner.TAG_1,
+            PowerPlayAutoRoadRunner.TAG_2,
+            PowerPlayAutoRoadRunner.TAG_3,
+        )
+        for (side in InitialSide.values()) {
+            for (tag in possibleTags) {
+                val set = buildTrajectories(side, drive, tag, side.startingPose())
+                trajectoryMap[TrajectoryParameters(side, tag)] = set
+                navigationTrajectoryMap[side] = Pair(set.navigateTrajectory, set.scoringPos)
+            }
+        }
+    }
+
+    fun getTrajectoryOne(side: InitialSide): Pair<TrajectorySequence, Pose2d> {
+        return navigationTrajectoryMap[side]!! // can be asserted nonnull because InitialSide is full in navigationTrajectoryMap
+    }
+
+    fun getTrajectory(side: InitialSide, foundTag: Int): TrajectorySet? {
+        return trajectoryMap[TrajectoryParameters(side, foundTag)]
+    }
+}
+
+private fun buildTrajectories(
+    side: InitialSide,
+    drive: SparkyMecanumDrive,
+    foundTag: Int,
+    start: Pose2d,
+): TrajectorySet {
+    var sequence = drive.trajectorySequenceBuilder(start)
+
+    val scoringPos: Pose2d
+    if (side === InitialSide.RIGHT) {
+        scoringPos = Pose2d(
+            PowerPlayAutoRoadRunner.SCOREPOS_RIGHT_X,
+            PowerPlayAutoRoadRunner.SCOREPOS_RIGHT_Y,
+            Math.toRadians(PowerPlayAutoRoadRunner.TURN_ONE_RIGHT)
+        )
+        sequence.splineToConstantHeading(
+            Vector2d(
+                34.0,
+                PowerPlayAutoRoadRunner.SIGNALPUSH_RIGHT_Y
+            ), Math.toRadians(90.0)
+        )
+        sequence.splineToConstantHeading(Vector2d(34.0, -12.0), Math.toRadians(90.0))
+        sequence.lineToSplineHeading(scoringPos)
+    } else {
+        scoringPos = Pose2d(
+            PowerPlayAutoRoadRunner.SCOREPOS_LEFT_X,
+            PowerPlayAutoRoadRunner.SCORE_POS_LEFT_Y,
+            Math.toRadians(PowerPlayAutoRoadRunner.TURN_ONE_LEFT)
+        )
+        sequence.splineToConstantHeading(
+            Vector2d(
+                -34.0,
+                PowerPlayAutoRoadRunner.SIGNALPUSH_LEFT_Y
+            ), Math.toRadians(90.0)
+        )
+        sequence.splineToConstantHeading(Vector2d(-34.0, -12.0), Math.toRadians(90.0))
+        sequence.lineToSplineHeading(scoringPos)
+    }
+    val trajectoryOne = sequence.build()
```

Comment:
After building `sequence`, I expect it to be discarded and a new builder to be used for the second sequence.  Even if it the builder object resets itself to be used for another trajectory, that's not an obvious behavior to the reader.